.github/workflows/build-and-quality-checks.yml

```
@@ -0,0 +1,24 @@
+name: Code Quality Checks
+on:
+  pull_request:
+    branches: ['v3.3']
+    types: ['opened', 'reopened', 'synchronize']
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: [ '1.19', '1.18', '1.17' ]
+    
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Go ${{ matrix.go }}
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Build Go ${{ matrix.go }}
+        run: go build -v ./...
+      - name: Test Go ${{ matrix.go }}
+        run: go test  . -v
```

To include tests in sub-directories:

```suggestion
        run: go test  ./... -v
```

---

There are no test cases in subdirectories. Do we need that?

---

future-proofing, if someone adds a sub-package he will not think of updating this yaml file

---

Cool.